artifactory/commands/transferconfig/transferconfigmerge.go

```diff
@@ -0,0 +1,373 @@
+				s = tcc.addToDifferentProperty(s, "Admin Privileges")
+			}
+		}
+		if s != "" {
+			conflict := tcc.newConflict(sourceProject.DisplayName, targetProject.DisplayName, sourceProject.ProjectKey, targetProject.ProjectKey, s)
+			conflicts = append(conflicts, conflict)
+			return conflicts, true
+		}
+
+	}
+	return conflicts, result
+}
+
+func (tcc *TransferConfigCommand) checkIfSameAdminPrivilige(source, target *accessServices.AdminPrivileges) bool {
+	if source == nil && target == nil {
+		return true
+	}
+
+	if source == nil || target == nil {
+		return false
+	}
+
+	// if source and target Admin priviliges are not nil then we have to check all pointer admin privilige have
+	manageMember := tcc.checkIfsameBoolPointer(source.ManageMembers, target.ManageMembers)
+	manageResouce := tcc.checkIfsameBoolPointer(source.ManageResources, target.ManageResources)
+	indexResouce := tcc.checkIfsameBoolPointer(source.IndexResources, target.IndexResources)
+	return (manageMember && manageResouce && indexResouce)
+
+}
+
+func (tcc *TransferConfigCommand) checkIfsameBoolPointer(source, target *bool) bool {
+	if source != nil && target != nil {
+		if *source != *target {
+			return false
+		}
+	}
+	if source == nil || target == nil {
+		if source != nil || target != nil {
+			return false
+		}
+	}
+	return true
+}
+
+func (tcc *TransferConfigCommand) addToDifferentProperty(s, property string) string {
+	if s == "" {
+		s = property
+		return s
+	}
+	s = s + ", " + property
+	return s
+}
+
+func (tcc *TransferConfigCommand) tryPing(targetServicesManager artifactory.ArtifactoryServicesManager) error {
+	_, err := targetServicesManager.Ping()
+	return err
+}
+
+func (tcc *TransferConfigCommand) validateMergeArtifactoryServers(targetServicesManager artifactory.ArtifactoryServicesManager, sourceArtifactoryVersion string, minRequiredVersion string) (bool, error) {
+	// if version is less than 7.0.0 projects are not defined, and we don't have to transfer them
+	transferProjects := true
+	if !version.NewVersion(sourceArtifactoryVersion).AtLeast(minRequiredVersion) {
+		transferProjects = false
+	}
+
+	// Avoid exporting and importing to the same server
+	log.Info("Verifying source and target servers are different...")
+	if tcc.sourceServerDetails.GetArtifactoryUrl() == tcc.targetServerDetails.GetArtifactoryUrl() {
+		return false, errorutils.CheckErrorf("The source and target Artifactory servers are identical, but should be different.")
+	}
+
+	// check correctness of Authorization
+	log.Info("Checking validation of your authorization methods..")
+	if tcc.tryPing(targetServicesManager) != nil {
+		return false, errorutils.CheckErrorf("The target's access token is not correct, please provide an availble access token.")
+	}
+	return transferProjects, nil
+}
+
+// Create a csv summary of all conflicts
+func (tcc *TransferConfigCommand) createConflictsCSVSummary(conflicts []ProjectConflict, timeStarted time.Time) (csvPath string, err error) {
```

Comment:
Ther fact that we have two functions which generate CSV files, one for projects and one for repositories, leads to inconsistencies. For example, the name of the files created is nor consistent. One includes "repositories" while the other does not include "projects". Let's try merging the two functions using an interface as an argument, and having the report name as an argument.
In case merging the functions proves impossible, let's try making the two consistent.

Reply:
I will ask Michael to do this tomorrow